lib/rules/aria-allowed-attr-matches.js

```diff
@@ -1,16 +1,9 @@
-var role = node.getAttribute('role');
-if (!role) {
-	role = axe.commons.aria.implicitRole(node);
-}
-var allowed = axe.commons.aria.allowedAttr(role);
-if (role && allowed) {
-	var aria = /^aria-/;
-	if (node.hasAttributes()) {
-		var attrs = node.attributes;
-		for (var i = 0, l = attrs.length; i < l; i++) {
-			if (aria.test(attrs[i].name)) {
-				return true;
-			}
+const aria = /^aria-/;
+if (node.hasAttributes()) {
+	let attrs = node.attributes;
+	for (let i = 0, l = attrs.length; i < l; i++) {
+		if (aria.test(attrs[i].name)) {
```

Comment:
Sorry, should have noticed this in my first review. It seems like you're not filtering out elements with only invalid ARIA attributes:

```
<input aria-invalidattr="foo" />
```

This element shouldn't be applicable to this rule. It's a pretty minor bug, so I'll leave it for you to decide if you want to bother with it. The fix is to use `axe.commons.aria.validateAttr` in the if statement here.